URL input polish

jasmussen opened the pull request:
This PR fixes two issues with the URL input dialog.

The first issue is that the URL bar, in very tight circumstances, would create a horizontal scrollbar — because it was abs positioned at the center of where it needed to be and then pulled leftwards 50% — but the invisible div was still poking out at the right. First commit fixes this.

The second was a scoping issue with the URL input when used for the Image block, when toolbar is fixed to top. This fixes that:

<img width="458" alt="screen shot 2018-04-20 at 10 17 36" src="https://user-images.githubusercontent.com/1204802/39038804-9fda72ec-4484-11e8-8b0a-41f67dc7f6b5.png">

Commit:
Fix horizontal scrollbar on url input

This fixes an issue where on small viewports, the url input box caused a horizontal scrollbar. This was because the parent div was abs positioned, and a child div was then horizontally pulled 50% to the left. By applying this left-pulling to the same div as is abs positioned, we ensure that the width of the url box is the same as what's visible.

Commit:
Fix issue with URL bar on image block with fixed toolbar

The Image URL box was broken if you had "fix toolbar to top" enabled. This fixes it.

noisysocks commented on blocks/rich-text/format-toolbar/index.js:
```diff
@@ -184,7 +184,7 @@ class FormatToolbar extends Component {
 				{ ( isAddingLink || isEditingLink || formats.link ) && (
 					<Fill name="RichText.Siblings">
-						<div style={ { position: 'absolute', ...focusPosition } }>
+						<div style={ { position: 'absolute', transform: 'translateX( -50% )', ...focusPosition } }>
```
One sneaky `position` was OK, but now that there are a few properties I think we ought to give this div a `className` and add a new rule to `format-toolbar/style.scss`:

```scss
.blocks-format-toolbar__link-container {
    position: absolute;
    transform: translateX( -50% );
}
```

noisysocks commented on blocks/rich-text/format-toolbar/style.scss:
```diff
@@ -14,6 +14,12 @@
+
+	.edit-post-header-toolbar__block-toolbar & {
+		position: absolute;
+    	left: 0;
```
There's some weird indentation going on here.

noisysocks reviewed (changes requested):
Thanks for fixing this! ❤️ 

The UI shown when adding a link to an image looks and behaves 💯.

I do see one regression, though. When inserting a link into a paragraph block, the UI moves around from side to side a lot while you are typing:

![links](https://user-images.githubusercontent.com/612155/39101392-61549552-46dc-11e8-8c4c-f8f342984933.gif)

Commit:
Address feedback.

This cleans things up, fixes a few typos, and addresses feedback by @noisysocks.

It also fixes a z-index issue where the suggestions would be below blocks.

jasmussen commented:
Thank you for the feedback! I believe I addressed it:

<img width="397" alt="screen shot 2018-04-23 at 11 59 58" src="https://user-images.githubusercontent.com/1204802/39120363-8497623a-46ee-11e8-9f5a-746b4ed8fa2e.png">

jasmussen commented on blocks/rich-text/format-toolbar/style.scss:
```diff
@@ -17,11 +16,17 @@
 }
 
+.blocks-format-toolbar__link-container {
+    position: absolute;
```
Hmm interesting — I was sure I installed eslint in VS Code — wondering why it didn't catch this...

Commit:
Spaces to tabs.

jasmussen commented:
Hooray! Thanks!
